### ISSUE TITLE
How to handle :symbol values?

### ISSUE THREAD

**Issue by matthewrudy** [activeadmin/activeadmin#3914]
We use [simple_enum](https://github.com/lwe/simple_enum) for enums.

Example

``` ruby
class User < ActiveRecord::Base
  as_enum :gender, female: 1, male: 0
end
```

But using this with the latest active admin
``` ruby
show do
  row "gender", :gender
end
```

This raises an exception `can't define singleton` (or something like that).

The key causes is `pretty_format` that has no implementation for symbols.

``` ruby
def pretty_format(object)
  case object
  when String, Numeric, Arbre::Element
    object.to_s
  when Date, Time
    localize object, format: :long
  else
    if defined?(::ActiveRecord) && object.is_a?(ActiveRecord::Base) ||
       defined?(::Mongoid)      && object.class.include?(Mongoid::Document)
      auto_link object
    else
      display_name object
    end
  end
end
```

My solution was to hack it;

``` ruby
module ActiveAdmin::ViewHelpers::DisplayHelper
  alias :pretty_format_before_symbol_hack :pretty_format
  def pretty_format(object)
    if object.is_a?(Symbol)
      object.to_s
    else
      pretty_format_before_symbol_hack(object)
    end
  end
end
```
But a nicer solution would be welcome :)

**Comment by timoschilling** [comment]
There is something on the road, take a look on the [pretty_format_api](https://github.com/activeadmin/activeadmin/tree/idea/pretty_format_api) branch.

**Comment by matthewrudy** [comment]
Ah, great.
So we'd just register a formatter.

That's a good solution.

**Comment by adambrod** [comment]
Same thing started failing for me on activeadmin 1.0.0.pre1 when we upgraded to ruby 2.2. I'll put in the stack trace to help others track this down. Using the hack from @matthewrudy gives us a workaround. Thanks!

```
 Failure/Error: click_on "Update Retailer"
     ActionView::Template::Error:
       can't define singleton
     # ./vendor/bundle/ruby/2.2.0/gems/activeadmin-1.0.0.pre1/lib/active_admin/view_helpers/method_or_proc_helper.rb:86:in `instance_exec'
     # ./vendor/bundle/ruby/2.2.0/gems/activeadmin-1.0.0.pre1/lib/active_admin/view_helpers/method_or_proc_helper.rb:86:in `render_in_context'
     # ./vendor/bundle/ruby/2.2.0/gems/activeadmin-1.0.0.pre1/lib/active_admin/view_helpers/display_helper.rb:18:in `display_name'
     # ./vendor/bundle/ruby/2.2.0/gems/activeadmin-1.0.0.pre1/lib/active_admin/view_helpers/display_helper.rb:55:in `pretty_format'
     # ./vendor/bundle/ruby/2.2.0/gems/arbre-1.0.3/lib/arbre/element.rb:180:in `method_missing'
     # ./vendor/bundle/ruby/2.2.0/gems/activeadmin-1.0.0.pre1/lib/active_admin/views/components/attributes_table.rb:82:in `content_for'
     # ./vendor/bundle/ruby/2.2.0/gems/activeadmin-1.0.0.pre1/lib/active_admin/views/components/attributes_table.rb:39:in `block (3 levels) in row'
     # ./vendor/bundle/ruby/2.2.0/gems/arbre-1.0.3/lib/arbre/element/builder_methods.rb:31:in `block in build_tag'
     # ./vendor/bundle/ruby/2.2.0/gems/arbre-1.0.3/lib/arbre/context.rb:92:in `with_current_arbre_element'
     # ./vendor/bundle/ruby/2.2.0/gems/arbre-1.0.3/lib/arbre/element/builder_methods.rb:49:in `with_current_arbre_element'
     # ./vendor/bundle/ruby/2.2.0/gems/arbre-1.0.3/lib/arbre/element/builder_methods.rb:26:in `build_tag'
     # ./vendor/bundle/ruby/2.2.0/gems/arbre-1.0.3/lib/arbre/element/builder_methods.rb:39:in `insert_tag'
     # ./vendor/bundle/ruby/2.2.0/gems/arbre-1.0.3/lib/arbre/element/builder_methods.rb:14:in `td'
     # ./vendor/bundle/ruby/2.2.0/gems/activeadmin-1.0.0.pre1/lib/active_admin/views/components/attributes_table.rb:38:in `block (2 levels) in row'
     # ./vendor/bundle/ruby/2.2.0/gems/activeadmin-1.0.0.pre1/lib/active_admin/views/components/attributes_table.rb:37:in `each'
     # ./vendor/bundle/ruby/2.2.0/gems/activeadmin-1.0.0.pre1/lib/active_admin/views/components/attributes_table.rb:37:in `block in row'
     # ./vendor/bundle/ruby/2.2.0/gems/arbre-1.0.3/lib/arbre/element/builder_methods.rb:31:in `block in build_tag'
     # ./vendor/bundle/ruby/2.2.0/gems/arbre-1.0.3/lib/arbre/context.rb:92:in `with_current_arbre_element'
```

**Comment by timoschilling** [comment]
closed by #4023
